### DISCUSSION TITLE
Widgets API

### DISCUSSION THREAD

**Discussion by kingofthebongo2008** [nxrighthere/UnrealCLR#55] [Help]
Hey,
I want to access the User Interface widgets in UnrealCLR.

Is there a way now? What should i do?
I have a textblock, that i want to change.

thanks

**Comment by kingofthebongo2008** [comment]
Actually my question now is how do i change a variable in a blueprint.
I have exposed the textblock as a variable.

public static void TestBlueprintComponentMethod(ObjectReference self)
        {
            string test = "Test";

            Object.GetText(self, "Text", test);
            //SceneComponent blueprintSceneComponent = self.ToComponent<SceneComponent>();
            //Debug.AddOnScreenMessage(-1, 30.0f, Color.Orange, "Cheers from managed function of the " + self.Name);
        }

I have looked around what to expose for the codegen, but some initial guideline will be fine.
Thank you.

**Comment by nxrighthere** [comment]
API of UI is not exposed yet (see https://github.com/nxrighthere/UnrealCLR/issues/38).

To get an idea of how to change blueprint properties see [this](https://github.com/nxrighthere/UnrealCLR/blob/master/Source/Managed/Tests/BlueprintsExtensibility.cs) test. Actors and components provide appropriate functions for that.

Modifications of codegen or framework internals by a user are not supported.

**Comment by kingofthebongo2008** [comment]
Thank you, but this test has scenecomponent and actors and i do not have such.
I have a widget with a text. 
What should i do to expose the widgets or least one of them?
I am ready to dig deeper if needed.

**Reply by nxrighthere** [reply]
Can you please share a screenshot or a C++ class of a widget that you are referring?

**Reply by kingofthebongo2008** [reply]
Can i change a widget text with an actor?

![2020-10-22 (3)](https://user-images.githubusercontent.com/1145201/96918393-e5f98000-14b2-11eb-92db-406b684b048c.png)

**Reply by nxrighthere** [reply]
This use case is valid, but the framework itself lacks the appropriate API. I'll take a look at what I can expose for you.

**Reply by kingofthebongo2008** [reply]
Yes, thank you.
Here i have marked a small textblock that must take a value from C#

![2020-10-22 (4)](https://user-images.githubusercontent.com/1145201/96918822-910a3980-14b3-11eb-94a8-7ec110d209cf.png)

**Reply by nxrighthere** [reply]
Yes, another problem is that the workflow with blueprints very limited at the moment (see https://github.com/nxrighthere/UnrealCLR/issues/33), but for your use case, I can expose a few things for now.
